### PR TITLE
added eth_ecrecover function:

### PR DESCRIPTION
**Pull request by agnescameron**
Added a function that allows the recovery of a signer's ethereum address from a signature created with ecsign (this method is already in the repo). ecrecover is also already used as a test for ecsign (in tests/requests.js), and this implementation (and the test) takes the same format. 

Commits were messy first time around -- tidied it up for clarity. Feedback appreciated + thanks!

### CODE REVIEW

**Comment by Firescar96** [comment]
@benjamincburns could you try out this PR? I cloned the repo locally and didn't get this errror.

**Comment by agnescameron** [comment]
@benjamincburns could you take a look? thanks

**Comment by benjamincburns** [comment]
@agnescameron & @Firescar96 sorry for the delay.

@agnescameron by any chance did you review trufflesuite/ganache-cli#243 when creating this PR?

File: test/requests.js [hunk 1]
```diff
@@ -382,6 +382,56 @@ var tests = function(web3) {
 
   });
 
+  describe("eth_ecRecover", function() {
```
Comment (benjamincburns) [inline test/requests.js]: Given that this isn't a part of the RPC standard, I wonder if this wouldn't be a better contribution to [truffle-contract](https://github.com/trufflesuite/truffle-contract) and/or [drizzle](https://github.com/trufflesuite/drizzle)?

Comment (benjamincburns) [reply test/requests.js]: I'm referring specifically to the logic contained in this function - obviously the other changes would still be necessary for that to work.

**Comment by Firescar96** [comment]
@benjamincburns bump

File: test/requests.js (hunk 1 above)
Comment (Firescar96) [reply test/requests.js]: I think this should be personal_ecRecover and there should be a corresponding personal_sign method, is that right?

**Comment by benjamincburns** [comment]
Hi @agnescameron!

Per @Firescar96 this should actually follow the geth management API and be called [`personal_ecRecover`](https://github.com/ethereum/go-ethereum/wiki/Management-APIs#personal_ecrecover) and there should be a [`personal_sign`](https://github.com/ethereum/go-ethereum/wiki/Management-APIs#personal_sign) to match.

These APIs are nonstandard, however they're implemented by both geth and Pairty.

If you can make those changes please we'll happily accept this PR.

**Comment by seesemichaelj** [comment]
Moving this to our backlog as we can implement the requested changes. If you feel the desire to complete this @agnescameron and no one assigns themselves to this PR, please make a comment and we'll let you make the changes. Thanks!

**Comment by davidmurdoch** [comment]
Closing as this PR is rather stale now.

**Pull request closed without merging**